### PR TITLE
Implement negation

### PR DESCRIPTION
**Pull request by Ladsgroup**
- Changing the state to accept negation in its mutation, getter, and
   actions.
 - Using the negation toggle component and make it send the change to
   state (and read from it, duh)
 - Rework state tests to have a base RootState and change that so for
   every change of state we wouldn't need to do the change in every case
 - Add tests for state changes and query condition using the negation

Bug: T272014
Bug: T272015

### CODE REVIEW

File: src/components/QueryCondition.vue [hunk 1]
```diff
@@ -5,10 +5,10 @@
 			:disabled="!canDelete"
 			@click="removeCondition"
 		/>
-		<div class="query-condition__toggle-button-group">
-			<Button disabled="true" style="border-right: 1px darkgrey solid;">with</Button>
-			<Button disabled="true">without</Button>
-		</div>
+		<NegationToggle
+			class="query-condition__toggle-button-group"
```
Comment (micgro42) [inline src/components/QueryCondition.vue:9]: We not using this class anywhere, right? If so, then we can just drop it

File: src/components/QueryCondition.vue [hunk 2]
```diff
@@ -122,6 +123,24 @@ export default Vue.extend( {
 				this.$store.dispatch( 'updateValue', { value, conditionIndex: this.conditionIndex } );
 			},
 		},
+		negate: {
```
Comment (micgro42) [inline src/components/QueryCondition.vue]: You can do that with one computed property. Having those two seems a bit too much

File: tests/unit/store/getters.spec.ts [hunk 3]
```diff
@@ -3,97 +3,53 @@ import getters from '@/store/getters';
...
-					propertyValueRelationData: { value: PropertyValueRelation.Matching },
-					conditionId: '0.123',
-					subclasses: false,
-				} ],
-				omitLabels: true,
-				errors: [],
-				limit: 0,
-				useLimit: false,
-			};
+			const state: RootState = JSON.parse( JSON.stringify( simpleRootState ) );
```
Comment (micgro42) [inline tests/unit/store/getters.spec.ts:31]: That could be done nicer with a method `getFreshRootState(): RootState`, but that can be refactored later. This is already an improvement :pray:

**Review by micgro42 — CHANGES_REQUESTED** [review]
Good start so far. In addition to the two points below, we can also make the `negate` field non-optional in the QueryRepresentation

File: src/components/QueryCondition.vue (hunk 1 above)
Comment (Ladsgroup) [reply src/components/QueryCondition.vue:9]: It's used (at least for now, until we use the wikit component, look at `&__toggle-button-group` line 190.

Comment (micgro42) [reply src/components/QueryCondition.vue:9]: Narf! I totally missed that! Cool, then that class is fine right now. Its naming is not ideal, but that can be changed in another PR, not this one.

File: src/components/QueryCondition.vue (hunk 2 above)
Comment (Ladsgroup) [reply src/components/QueryCondition.vue]: Yeah, it grew organically when I was writing the code. Fixed now.

**Comment by Ladsgroup** [comment]
> d start so far. In addition to the two points below, we can also make the `negate` field non-optional in the QueryRepresentation

I just tried it but buildQuery.js tests exploded, needs a refactor that I'll do separately with the `getFreshRootState()` refactor.

File: src/components/QueryCondition.vue (hunk 2 above)
Comment (micgro42) [reply src/components/QueryCondition.vue]: :+1:

**Review by micgro42 — APPROVED** [review]

**Pull request merged**